src/cli/utils/getRoute.ts

```diff
@@ -12,28 +21,43 @@ function removeGroupSegments(input: string) {
 }
 
 function removeParallelSegments(input: string) {
-  return input.replace(/\/@[\w]+/g, '')
+  return input.replace(/\/@\w+/g, '')
+}
+
+function removeInterceptedSegments(input: string) {
+  let result = input.replace(/\(\.\)/g, '')
+  const twoDotsRegExp = /[[\w-\]]+\/\(\.{2}\)/g
+  while (twoDotsRegExp.test(result)) {
+    result = result.replace(twoDotsRegExp, '')
+  }
+  return result.replace(/.*\(\.{3}\)/g, '/')
 }
 
 function formatDynamicSegments(input: string) {
   return input.replace(/\/\[(\w+)\]/g, '/:$1')
 }
 
 function getRouteName({ originPath }: Rewrite) {
-  const formatRouteName = pipe(removePageSegment)
+  const formatRouteName = pipe(
+    removePageSegment,
+    removeGroupSegments,
+    removeParallelSegments,
+    removeInterceptedSegments
+  )
   return asRootPath(formatRouteName(originPath))
 }
 
 function getRouteHref({ localizedPath }: Rewrite) {
+  const localeSegment = extractLocaleSegment(localizedPath)
```

Comment:
Could you explain a bit why you need to extract locale here?

Reply:
I extracted it because the three dot `(...)`  version intercepts routes starting from the root level and I had to make sure that locale segment, for example `/en` is not removed. But now that I think of this, this can be implemented by writing a positive lookbehind for the part that removes parts before the `(...)` interception. 

I will simplify this with that regexp improvement.

Reply:
Oh yeah, the positive lookbehind version won't work with both getRouteHref and getRouteName functions, since with getRouteName we want to remove all segments before `(...)` part and with getRouteHref we want to keep the locale segment `/(en)`. That's why I created a separate function that extracts the locale segment in the getRouteHref function.

Reply:
If you think there is a better way to implement this, please guide me to the right direction 😄

Reply:
I see, thank you for explanation.